awscli/customizations/history/db.py

```diff
@@ -0,0 +1,256 @@
+# Copyright 2017 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+import os
+import uuid
+import time
+import json
+import base64
+import datetime
+import threading
+import logging
+from collections import MutableMapping
+
+from botocore.history import BaseHistoryHandler
+
+from awscli.compat import sqlite3
+from awscli.compat import ensure_text_type
+from awscli.compat import binary_type
+
+
+LOG = logging.getLogger('awscli.history.db')
+
+
+class DatabaseConnection(object):
+    _CREATE_TABLE = """
+    CREATE TABLE IF NOT EXISTS records (
+      id TEXT,
+      request_id TEXT,
+      source TEXT,
+      event_type TEXT,
+      timestamp INTEGER,
+      payload TEXT
+    )"""
+    _ENABLE_WAL = """PRAGMA journal_mode=WAL"""
+    _DATABASE_NAME = 'history.db'
+    _DEFAULT_DATABASE_FILENAME = os.path.expanduser(
+        os.path.join('~', '.aws', 'cli', 'history', _DATABASE_NAME))
+
+    def __init__(self, db_filename=None):
+        if db_filename:
+            self._db_filename = db_filename
+        elif os.environ.get('AWS_HISTORY_PATH'):
```

Comment:
I don't believe this was discussed.  I'd prefer to not add this env var for now.